frame/executive/src/lib.rs

```diff
@@ -315,8 +318,12 @@ where
 			"Parent hash should be valid.",
 		);
 
-		if let Err(i) = System::ensure_inherents_are_first(block) {
-			panic!("Invalid inherent position for extrinsic at index {}", i);
+
+		match System::ensure_inherents_are_first(block) {
+			Ok(i) => return i,
+			Err(i) => {
+				panic!("Invalid inherent position for extrinsic at index {}", i);
```

Comment:
I find it weird that the code panics here when an inherent extrinsic is not ordered properly, wouldn't we want to return a `Result` and let the caller of `initial_checks` handle the error gracefully instead of panicking?

Reply:
This is when executing a block, if the inherent are not first then the block is not valid, similar to when the signature are wrong. In those case we simply panic as the block is invalid.